include/mata/parser/mintermization.hh

```diff
@@ -17,42 +17,79 @@
 #ifndef MATA_MINTERM_HH
 #define MATA_MINTERM_HH
 
-#include "mata/cudd/cuddObj.hh"
-
 #include "inter-aut.hh"
+#include "bdd-domain.hh"
 
 namespace mata {
 
+template <class MintermizationDomain>
 class Mintermization {
 private: // data types
-    struct OptionalBdd {
-        enum class TYPE {NOTHING_E, BDD_E};
+    struct OptionalValue {
```

Comment:
Could we have some comments about what the class is and what its methods do? To facilitate their easier use.